In [3]:
from echo.multi_threading_agent_pydantic_second import (find_teams, create_value_prop)
from echo.multi_threading_agent_pydantic_second import get_company_from_url, get_people_data, enrich_people_data, get_influence_score, get_graphs_for_initiatives


import openai
import os
api_key = os.getenv("OPENAI_API_KEY")
client = openai.OpenAI(api_key=api_key)
#client = openai.OpenAI(api_key="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")

buyer = "rippling"
seller = "whatfix"

buyer_initiatives  = """
Buyer is rippling.com. Rippling is a San Francisco-based HR technology company founded in 2016 by Parker Conrad and Prasanna Sankar. It offers a unified platform that integrates HR, IT, and finance functions, enabling businesses to manage employee onboarding, payroll, benefits, device management, and compliance from a single system. The company targets firms with fewer than 2,000 employees and competes with established players like Workday and ADP, as well as startups like Justworks and Deel. As of April 2024,
Rippling achieved a valuation of $13.5 billion after a $200 million funding round led by Coatue, with participation from investors such as Founders Fund, Greenoaks, and Dragoneer .​
Top 3 Goals for 2025:
1. International Expansion: Rippling is focusing on growing its global presence, particularly in India. The company plans to hire over 100 engineers in Bengaluru and expand its workforce across various functions, including product management, design, sales, finance, and customer support .​
2. Investment in Research and Development (R&D): With over $1 billion in cash reserves, Rippling aims to invest heavily in R&D to develop new products and enhance its existing offerings. This includes building tools that support clients and integrating advanced technologies like artificial intelligence to improve workforce management solutions .​
3. Enhancing IT Security and Automation: Recognizing the growing importance of cybersecurity, Rippling is prioritizing IT security by addressing challenges such as data privacy, compliance, and automation of IT processes. The company is working on implementing robust controls, improving identity and device management solutions, and reducing manual, repetitive tasks to minimize human error and enhance overall security .​
These strategic goals align with Rippling's mission to provide a comprehensive, integrated platform that simplifies workforce management for businesses worldwide.​
"""

seller_info = """
​Whatfix is a global leader in digital adoption platforms (DAP),
co-headquartered in San Jose, California, and Bengaluru, India. Founded in 2014 by Khadim Batti and Vara Kumar,
the company offers solutions that enhance user engagement and productivity across web, desktop, and mobile applications.
Serving over 700 enterprises, including more than 80 Fortune 500 companies like Shell, Microsoft, and Schneider Electric,
Whatfix has established a strong presence in the digital transformation space .​


🚧 Key Challenges Addressed by Whatfix
Whatfix's platform is designed to tackle several common challenges faced by organizations:​
Complex Software Onboarding: Traditional onboarding processes can be time-consuming and ineffective. Whatfix provides in-app guidance tools—such as Flows, Task Lists, and Smart Tips—that help users navigate complex applications, reducing the learning curve and enhancing user adoption .​
High Training and Support Costs: Organizations often spend substantial resources on training and support. By offering self-help modules and contextual assistance within applications, Whatfix reduces the need for extensive training sessions and support interventions, leading to significant cost savings .​
Whatfix
Low Feature Adoption: New features in applications may go unnoticed or underutilized. Whatfix addresses this by delivering targeted, in-app prompts and walkthroughs that highlight new functionalities, ensuring users are aware of and can effectively use all features .​
Process Non-Compliance: Ensuring that employees follow standardized processes is crucial for operational efficiency. Whatfix's real-time guidance and analytics help monitor user interactions, ensuring compliance with established workflows and identifying areas for improvement .​
Celent
Inefficient Feedback Mechanisms: Gathering user feedback is essential for continuous improvement. Whatfix enables organizations to collect real-time feedback through in-app surveys and quizzes, providing insights into user experiences and training effectiveness .​
Whatfix

By addressing these challenges, Whatfix empowers organizations to enhance user engagement, streamline training processes, and maximize the return on their technology investments.

"""


In [2]:
seller = 'https://whatfix.com/'
buyer = 'https://rippling.com/'
from echo.indexing import create_tables
from echo.step_templates.utilities.seller_web_search import add_landing_page_data_to_seller_index, add_seller_web_search_data
from echo.utils import json_to_markdown
create_tables(seller)
from echo.step_templates.utilities.buyer_web_search import add_buyer_web_search_data





Enums file already exists.
Created tables for https://whatfix.com/


In [7]:
import json
def return_dummy_content_json():
    f = open("./echo/data/content.json",'r').read()
    return json.dumps(json.loads(f))
print(return_dummy_content_json())
print(type(return_dummy_content_json()))    


{"type": "doc", "content": [{"type": "heading", "attrs": {"textAlign": null, "level": 1}, "content": [{"type": "text", "text": "Getting started"}]}, {"type": "paragraph", "attrs": {"textAlign": null}, "content": [{"type": "text", "text": "Welcome to the "}, {"type": "text", "marks": [{"type": "italic"}, {"type": "highlight", "attrs": {"color": "var(--tt-color-highlight-yellow)"}}], "text": "Simple Editor"}, {"type": "text", "text": " template! This template integrates "}, {"type": "text", "marks": [{"type": "bold"}], "text": "open source"}, {"type": "text", "text": " UI components and Tiptap extensions licensed under "}, {"type": "text", "marks": [{"type": "bold"}], "text": "MIT"}, {"type": "text", "text": "."}]}, {"type": "paragraph", "attrs": {"textAlign": null}, "content": [{"type": "text", "text": "Integrate it by following the "}, {"type": "text", "marks": [{"type": "link", "attrs": {"href": "https://tiptap.dev/docs/ui-components/templates/simple-editor", "target": "_blank", "rel"

In [2]:
# flow
from echo.content_gen_agent import get_chunks_from_call_discovery
# seller research

#add_landing_page_data_to_seller_index(seller)
#add_seller_web_search_data(seller)


# buyer research
#buyer_data = add_buyer_web_search_data(buyer=buyer, seller=seller)
from echo.multi_threading_agent_pydantic import account_plan_extract_data
#final_data = account_plan_extract_data(buyer_data["full_text"], buyer, seller)
#print(final_data.model_dump())


# process discovery call

from echo.step_templates.discovery import BuyerDataExtracted

# fetch discovery transcript from constants or link 
from echo.step_templates.sample_data import SAMPLE_DISCOVERY_CALL_TRANSCRIPT

# fetch discovery parsing prompt


# run through discovery parsing using above pydantic
# create index
# how do we updates?
sample_transcript = """
AE: Thanks for joining today. Curious to hear how onboarding works at your org?

IT Lead (Sandeep): Yeah, it’s a bit of a mess honestly. IT has to manually provision every new tool. Takes us 2–3 days per person. 

AE: Got it. And what’s the ideal state?

Sandeep: Something more self-serve, like how Notion works. New hires should be able to get access without filing IT tickets.

Finance Lead (Aparna): For us, the concern is cost. We’ve already got too many tools eating into our budget. Unless there’s a clear ROI or consolidation angle, we’ll get pushback.

AE: Totally makes sense. What would success look like for you?

Aparna: If it saves us even one headcount worth of time, that’s meaningful.

Champion (Ravi): Let me get this in front of our CTO. I’ll probably share a short version of what you showed — maybe a doc or something? He’s most concerned about security and integration.

AE: Happy to help you shape that.

"""

chunks = get_chunks_from_call_discovery(buyer, seller,sample_transcript,call_id="Default")
print(chunks)




Enums file already exists.
{'chunks': [{'text': 'Manual provisioning of every new tool takes 2–3 days per person', 'persona_speaking': <StakeholderEnum.IT: 'IT'>, 'persona_about': <StakeholderEnum.IT: 'IT'>, 'type': 'pain_point', 'seller': 'Rippling/Whatfix', 'buyer': 'Prospect', 'call_id': 'Default'}, {'text': 'Desire for self-serve onboarding like Notion to eliminate IT ticket filing', 'persona_speaking': <StakeholderEnum.IT: 'IT'>, 'persona_about': <StakeholderEnum.IT: 'IT'>, 'type': 'goal', 'seller': 'Rippling/Whatfix', 'buyer': 'Prospect', 'call_id': 'Default'}, {'text': 'Concern about cost: too many tools impacting budget, need clear ROI or consolidation to avoid pushback', 'persona_speaking': <StakeholderEnum.Finance: 'Finance'>, 'persona_about': <StakeholderEnum.Finance: 'Finance'>, 'type': 'objection', 'seller': 'Rippling/Whatfix', 'buyer': 'Prospect', 'call_id': 'Default'}, {'text': 'Success if solution saves time equivalent to one headcount', 'persona_speaking': <Stakeholder

In [2]:
from echo.content_gen_agent import group_chunks_by_persona, create_chroma_collections
from echo.content_gen_agent import StakeholderEnum
chunks = {'chunks': [{'text': 'Manual provisioning of every new tool takes 2–3 days per person', 'persona_speaking': StakeholderEnum.IT, 'persona_about': StakeholderEnum.IT, 'type': 'pain_point', 'seller': 'Rippling/Whatfix', 'buyer': 'Prospect', 'call_id': 'Default'}, {'text': 'Desire for self-serve onboarding like Notion to eliminate IT ticket filing', 'persona_speaking': StakeholderEnum.IT, 'persona_about': StakeholderEnum.IT, 'type': 'goal', 'seller': 'Rippling/Whatfix', 'buyer': 'Prospect', 'call_id': 'Default'}, {'text': 'Concern about cost: too many tools impacting budget, need clear ROI or consolidation to avoid pushback', 'persona_speaking': StakeholderEnum.Finance, 'persona_about': StakeholderEnum.Finance, 'type': 'objection', 'seller': 'Rippling/Whatfix', 'buyer': 'Prospect', 'call_id': 'Default'}, {'text': 'Success if solution saves time equivalent to one headcount', 'persona_speaking': StakeholderEnum.Finance, 'persona_about': StakeholderEnum.Finance, 'type': 'goal', 'seller': 'Rippling/Whatfix', 'buyer': 'Prospect', 'call_id': 'Default'}, {'text': 'Champion will share a short summary doc with CTO, emphasizing security and integration', 'persona_speaking': StakeholderEnum.IT, 'persona_about': StakeholderEnum.Executive, 'type': 'champion_signal', 'seller': 'Rippling/Whatfix', 'buyer': 'Prospect', 'call_id': 'Default'}], 'buyer': 'Prospect', 'seller': 'Rippling/Whatfix', 'call_id': 'Default'}

grouped_chunks = group_chunks_by_persona(chunks['chunks'])
create_chroma_collections( buyer, seller, grouped_chunks)

print("Grouped Chunks:")
print(grouped_chunks)


# select top 4 sections or get template based on inputs
# query respective queries and subquery to get the doc

# get the data from the doc


Grouped Chunks:
defaultdict(<class 'list'>, {<StakeholderEnum.IT: 'IT'>: [{'text': 'Manual provisioning of every new tool takes 2–3 days per person', 'persona_speaking': <StakeholderEnum.IT: 'IT'>, 'persona_about': <StakeholderEnum.IT: 'IT'>, 'type': 'pain_point', 'seller': 'Rippling/Whatfix', 'buyer': 'Prospect', 'call_id': 'Default'}, {'text': 'Desire for self-serve onboarding like Notion to eliminate IT ticket filing', 'persona_speaking': <StakeholderEnum.IT: 'IT'>, 'persona_about': <StakeholderEnum.IT: 'IT'>, 'type': 'goal', 'seller': 'Rippling/Whatfix', 'buyer': 'Prospect', 'call_id': 'Default'}], <StakeholderEnum.Finance: 'Finance'>: [{'text': 'Concern about cost: too many tools impacting budget, need clear ROI or consolidation to avoid pushback', 'persona_speaking': <StakeholderEnum.Finance: 'Finance'>, 'persona_about': <StakeholderEnum.Finance: 'Finance'>, 'type': 'objection', 'seller': 'Rippling/Whatfix', 'buyer': 'Prospect', 'call_id': 'Default'}, {'text': 'Success if solutio

In [2]:
# query index

from echo.content_gen_agent import query_stakeholder_index, StakeholderEnum
ans = query_stakeholder_index(
        buyer,seller,
    persona=StakeholderEnum.IT)
print("Query Result for IT Persona:")
print(ans)  

Number of requested results 5 is greater than number of elements in index 2, updating n_results = 2


Query Result for IT Persona:
[{'text': 'Desire for self-serve onboarding like Notion to eliminate IT ticket filing', 'type': 'goal', 'source': None}, {'text': 'Manual provisioning of every new tool takes 2–3 days per person', 'type': 'pain_point', 'source': None}]


In [4]:
# dummy func 1
from echo.content_gen_agent import query_stakeholder_index, StakeholderEnum
from echo.constants import DEMO
from echo.step_templates.prompts import DEMO_BUSINESS_CASE_CONTENT_CHAMPION
def get_content_doc(buyer, seller, doc_type="Business_Case"):
    api_key = os.getenv("OPENAI_API_KEY")
    client = openai.OpenAI(api_key=api_key)

    


    business_case_user_prompt = f"""    
    the buyer is {buyer} and the seller is {seller}.
    """
    

    response_team_guess_reasoning = client.responses.create(
        model="o4-mini",
        reasoning={"effort": "medium"},
        input=[
            {"role": "system", "content": DEMO_BUSINESS_CASE_CONTENT_CHAMPION},
            {"role": "user", "content": business_case_user_prompt},
        ],
    )
    #print(response_team_guess_reasoning)
    return response_team_guess_reasoning
    
#print(DEMO_BUSINESS_CASE_CONTENT_CHAMPION)
a = get_content_doc(buyer, seller, doc_type="Business_Case")
print(a.output_text)

# func 2
from echo.step_templates.prompts import ASK_AI_SYSTEM_PROMPT
def get_ai_answers(buyer, seller, query, stakeholder, department, doc_context):
    api_key = os.getenv("OPENAI_API_KEY")
    client = openai.OpenAI(api_key=api_key)

    

    response_team_guess_reasoning = client.responses.create(
        model="o4-mini",
        reasoning={"effort": "medium"},
        input=[
            {"role": "system", "content": ASK_AI_SYSTEM_PROMPT},
            {"role": "user", "content": query},
        ],
    )
    #print(response_team_guess_reasoning)
    return response_team_guess_reasoning

query = """
What timelines are we looking at for roi?
"""
stakeholder = "IT Lead"
department = StakeholderEnum.IT
doc_context = a.output_text
b = get_ai_answers(buyer, seller, query, stakeholder, department, doc_context)
print(b.output_text)





# DUMMY FUNC 3
from echo.step_templates.prompts import OBJECTION_PREDICTION_SYSTEM_PROMPT
def get_objection_handling_response(buyer, seller, stakeholder, department, doc_context):
    api_key = os.getenv("OPENAI_API_KEY")
    client = openai.OpenAI(api_key=api_key)

    objection_handling_user_prompt = f"""
    The buyer is {buyer} and the seller is {seller}.
    the department is {department}. and the stakeholder is {stakeholder}.
    here is the context of document the buyer is seeing - {doc_context}
    """
    response_team_guess_reasoning = client.responses.create(
        model="o4-mini",
        reasoning={"effort": "medium"},
        input=[
            {"role": "system", "content": OBJECTION_PREDICTION_SYSTEM_PROMPT},
            {"role": "user", "content": objection_handling_user_prompt},
        ],
    )
    #print(response_team_guess_reasoning)
    return response_team_guess_reasoning
c = get_objection_handling_response(buyer,seller,stakeholder, department, doc_context)
print(c.output_text)

# High – avoid continued onboarding delays by Q4 HR system launch

*Developed by*: **Alice Chen**, with **IT Ops, HR, Finance heads**

---

**Headline**

We will "streamline employee onboarding", remove "fragile manual processes", build a "unified workforce platform", and "reduce time to productivity" to eliminate "onboarding delays". By "streamline employee onboarding", "reduce time to productivity", and our "unified workforce platform", we cut "fragile manual processes", stop "onboarding delays", and "streamline employee onboarding" to "reduce time to productivity".

---

### The Problem Statement

Each week new hires face a 2-hour first-day lag costing $150 per person (source – Rippling HRIS). If unaddressed by Q4 HR system launch, fragmentation worsens.

---

### Recommended Approach

1. Contextual in-app guidance tackles halted tasks  ➜ [Case Study – ACME](https://www.notion.so/abitnikunj/case_acme)  
2. No-code editor in the "unified workforce platform" cuts IT dependency  ➜ [G2 

In [3]:
#a.output_text
b.output_text

'Here’s what we typically see in engagements like yours—of course, we’ll refine this once we know your exact scope, baseline metrics and go-live cadence:\n\n• Pilot / proof-point (months 1–3)  \n  – Set up core modules  \n  – Begin tracking your first efficiency gains (e.g. reduced manual effort, faster cycle-times)  \n\n• Early ROI (months 4–6)  \n  – Broader roll-out to remaining teams/functions  \n  – Measurable cost savings or revenue uplift (often 10–15% on targeted processes)  \n\n• Break-even / payback (months 9–12)  \n  – Cumulative benefits cover your full implementation and subscription costs  \n  – You’ll have solid data to project ongoing annual savings or incremental revenue  \n\n• Full-value realization (months 12–18)  \n  – Optimization, advanced analytics and additional modules drive further upside (20–30%+ gains)  \n\nTo build a tailored model for you, could you share:  \n1) Which modules or features you plan to deploy first  \n2) Your current process benchmarks (e.g. 

# High Priority – avoid compliance violations and payroll errors by year-end audit

*Developed by*: **Jordan Lee**, with **IT Operations, HR, Finance**

---

**Headline**

Because “unifies HR, IT, and Finance operations on a single, modern platform” is reshaping “global workforce management”, we should “simplify workforce management” by implementing Whatfix in Q3 2025. Doing so avoids compliance violations and payroll errors and unlocks “single source of truth” and “automation”.

---

### The Problem Statement

Every month, global HR and finance teams experience manual steps cause delays and errors, costing $250K in rework and compliance fines (*source – internal audit*). If unaddressed by year-end audit, error rates will increase by 15%.

---

### Recommended Approach

- Pain ➜ Capability ➜ Proof  
  1. manual steps cause delays and errors ➜ contextual in-app guidance ➜ [Case Study: Acme Corp](https://whatfix.com/case-study)  
  2. complex software workflows ➜ interactive walkthroughs ➜ [Customer Testimonial](https://whatfix.com/testimonial)  
  3. Sandbox rollout & multilingual support ensure adoption across global HR and finance teams  
• Success factors: Cross-functional team of 3: IT lead, HR lead, Finance analyst; SSO integration; store-level champions

---

### Target Outcomes

| Metric                | Current | Target by year-end audit | Δ     |
|-----------------------|---------|--------------------------|-------|
| Time to proficiency   | 5 days  | 2 days                   | –3 d  |
| Error rate            | 5 %     | 1 %                      | –4 %  |
| Employee satisfaction | 70 %    | 85 %                      | +15 % |

*(Δ = Target − Current when both are numeric.)*

---

### Required Investment

- **Budget**: USD 120 K annual subscription – per user per month $25; payback in < 6 mo.  
- **People & Time**: Cross-functional team of 3: IT lead, HR lead, Finance analyst. Kickoff Q3 2025.  
- **Next Step**: approve charter this week to meet year-end audit.

---

### Evidence & Compliance

*SOC-2 Certificate*  
*ISO 27001 Compliance*
